File: 0594-Longest Harmonious Subsequence/0594-Longest Harmonious Subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        int maxCount = 0;
        unordered_map<int, int> table;
        for (int num : nums)
        {
            int count = ++table[num];
            if (table.find(num - 1) != table.end())
            {
                maxCount = max(maxCount, count + table[num - 1]);
            }
            if (table.find(num + 1) != table.end())
            {
                maxCount = max(maxCount, count + table[num + 1]);
            }
        }
        
        return maxCount; 
    }
};

File: 0501-0600/0594-Longest Harmonious Subsequence/0594-Longest Harmonious Subsequence.py
import collections
class Solution:
    def findLHS(self, nums: List[int]) -> int:
        table = collections.Counter(nums)
        maxCount = 0
        for num in table:
            if num + 1 in table:
                maxCount = max(maxCount, table[num] + table[num + 1])
        return maxCount
